### PR TITLE
`Development`: Fix some constantly failing Postgres tests and LinkPreviewIntegrationTest

### PR DESCRIPTION
**Pull request by reschandreas**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context

This PR fixes some postgres tests because an invalid SQL construct was used. It also fixes the now constantly failing LinkPreviewIntegrationTest which should be completely refactored, I'll open an issue -> #8054

The postgres tests still don't pass completely, one to two tests are very flacky and I can't reproduce that on my machine.

### Steps for Testing

Check if the SQL has changed, because it should not

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **Refactor**
	- Improved query efficiency and accuracy in the results repository.
- **Tests**
	- Updated integration tests to reflect changes in referenced URLs, enhancing test reliability.
- **Chores**
	- Updated import statements and domain class list in the ResultRepository.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java [hunk 1]
```diff
@@ -5,15 +5,29 @@
...
-import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.ExampleSubmission;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.Feedback;
+import de.tum.in.www1.artemis.domain.GradingCriterion;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.Submission;
+import de.tum.in.www1.artemis.domain.User;
```
[the remark this answers is not in the document]
Comment (reschandreas) [reply src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java:30]: no we changed that, now we want to get rid of the star imports

**Review by MichaelOwenDyer — APPROVED** [review]
Thank you for fixing this!

File: src/test/java/de/tum/in/www1/artemis/metis/linkpreview/LinkPreviewIntegrationTest.java [hunk 2]
```diff
@@ -24,8 +24,7 @@ class LinkPreviewIntegrationTest extends AbstractSpringIntegrationIndependentTes
     // this link will return null for all fields because it does not include OG tags
     private static final String GOOGLE_URL = "https://google.com";
```
Comment (rstief) [inline src/test/java/de/tum/in/www1/artemis/metis/linkpreview/LinkPreviewIntegrationTest.java:26]: this still calls a foreign url

Comment (reschandreas) [reply src/test/java/de/tum/in/www1/artemis/metis/linkpreview/LinkPreviewIntegrationTest.java:26]: all of them do, I tried it with artemis URLs but then the test failed, I hope someone fixes this when someone takes on #8054 as I don't want to loose much time right now

**Review by laurenzfb — APPROVED** [review]
✅ ✅ ✅

**Review by milljoniaer — APPROVED** [review]
Thanks for fixing this 🍏

**Comment by sillyhong** [comment]
@coderabbitai please review

**Comment by krusche** [comment]
> @coderabbitai please review

why?

**Comment by omar-coderabbitai** [comment]
@coderabbitai configuration

**Pull request merged**